lib/sse.js

```diff
@@ -184,7 +184,6 @@ var SSE = function (url, options) {
     this.xhr.addEventListener('load', this._onStreamLoaded.bind(this));
     this.xhr.addEventListener('readystatechange', this._checkStreamClosed.bind(this));
     this.xhr.addEventListener('error', this._onStreamFailure.bind(this));
-    this.xhr.addEventListener('abort', this._onStreamFailure.bind(this));
```

Comment:
Not calling `_onStreamFailure()` means you're no longer going through the `close()` method that cleans up and resets the internal state.

Reply:
I am using the close method in my code where I'm using it. The `close()` method calls the `xhr.abort()`. Let's say a user action initiated a SSE connection & then after some time, user chose to cancel/close the connection. Should that fire an error event? That's more like a user action to me. 

There's another problem because of this. If an error chunk comes in the SSE and it does trigger the regular 'error' event. Let's say we've configured it to close the sse connection in that case. That causes the same error event to be fired again. Let's say we've configured it to show a snackbar whenever an error event gets fired. In this case the error snackbar would show up twice & user would wonder if 2 errors happened.

So to resolve this, I'd say let's make sure either there's a way to distinguish between the two events (error & close/abort) OR the error custom event gets emitted only when xhr error happens.

Reply:
I agree; I would recommend having a `_onStreamAbort()` maybe that sends an abort custom event, and cleanly closes things out like `_onStreamFailure()` does.

Reply:
Cool. I just checked in changes to this PR with the `_onStreamAbort()` method. Can you review/approve?